ims/einstein/iscsi/tgt.py

```diff
@@ -125,14 +125,19 @@ def remove_target(self, target_name):
         try:
             targets = self.list_targets()
             if target_name in targets:
-                os.remove(os.path.join(self.TGT_ISCSI_CONFIG,
-                                       target_name + ".conf"))
                 command = "tgt-admin -f --delete {0}".format(target_name)
                 output = shell.call(command, sudo=True)
                 logger.debug("Output = %s", output)
+                os.remove(os.path.join(self.TGT_ISCSI_CONFIG,
+                                       target_name + ".conf"))
             else:
-                raise iscsi_exceptions.TargetDoesntExistException()
-        except (IOError, OSError) as e:
+                logger.info("%s target doesnt exist" % target_name)
+        except OSError as e:
+            if "[Errno 2] No such file or directory" in str(e):
```

Comment:
you should probably compare the value of `e.errno` to `2`, or for bonus points, compare it to `os.ENOENT`